include/libdnf/base/base.hpp

```diff
@@ -90,24 +89,23 @@ class Base {
     libdnf::module::ModuleItemContainerWeakPtr get_module_item_container() {
         return module_item_container.get_weak_ptr();
     }
-    libdnf::advisory::AdvisorySackWeakPtr get_rpm_advisory_sack() { return rpm_advisory_sack.get_weak_ptr(); }
 
     /// Gets base variables. They can be used in configuration files. Syntax in the config - ${var_name} or $var_name.
     VarsWeakPtr get_vars() { return VarsWeakPtr(&vars, &vars_gurad); }
 
-    /// @return The system state object.
-    /// @since 5.0
-    // TODO(lukash) if it stays on public API, use WeakPtr
-    libdnf::system::State & get_system_state() { return *system_state; }
-
     void add_plugin(plugin::IPlugin & iplugin_instance);
     void load_plugins();
     plugin::Plugins & get_plugins() { return plugins; }
 
     libdnf::BaseWeakPtr get_weak_ptr() { return BaseWeakPtr(this, &base_guard); }
 
+    class Impl;
```

Comment:
What about to move `Impl` forward declaration above  the usage? It is like that in `PackageQuery`.

Reply:
This is a change from: https://github.com/rpm-software-management/dnf5/pull/18
It would be better to review those changes there.